missing include

It looks like there is a missing `#include <stdexcept>` in database_pimpl.hpp

I see the following when trying to build using gcc-11:

```
/home/kgasperich/work/nwx-workspace/nwchemex/build/_deps/pluginplay-src/src/pluginplay/database/detail_/database_pimpl.hpp:278:16: error: 'out_of_range' is not a member of 'std'
  278 |     throw std::out_of_range("Key was not found in the database");
      |                ^~~~~~~~~~~~
```

--- a/src/pluginplay/database/detail_/database_pimpl.hpp
+++ b/src/pluginplay/database/detail_/database_pimpl.hpp
@@ -1,5 +1,6 @@
 #pragma once
 #include "../value.hpp"
+#include <stdexcept>
 
 namespace pluginplay::database::detail_ {
 

